### PR TITLE
Add support for arbitrary conflict resolutions

### PR DESCRIPTION
**Pull request by olsen232**
```
$ sno conflicts -s --flat
table_A:fid=5
table_A:fid=6
table_A:fid=7

$ sno resolve table_A:fid=5 --with-file=5.geojson
Resolved 1 conflict. 2 conflicts to go.
```

The user can provide an arbitrary number of features in a geojson file, and store them as the resolution to a conflict. Generally, the appropriate number of features will be either 1 - in the case of an edit/edit conflict that needs manual resolution - or 2, in the case of an add/add conflict, the resolution should probably contain both features in some form. (Zero is also a useful number of features, but it is already possible to resolve a conflict by supplying zero features without providing a file).

### CODE REVIEW

File: sno/resolve.py [hunk 1]
```diff
@@ -1,53 +1,97 @@
...
+    return result
+
+
+def ungeojson_file(file_path, dataset):
+    """
+    Given a file containing multiple geojson features belonging to dataset,
+    returns the features as dicts containing gpkg geometries.
+    """
+    content = Path(file_path).read_text(encoding="utf-8")
+    features = json.loads(content)["features"]
```
Comment (craigds) [inline sno/resolve.py:29]: Use `json.load(Path(file_path))` to lower memory usage here

File: sno/resolve.py [hunk 2]
```diff
@@ -1,53 +1,97 @@
...
+
+    features = ungeojson_file(file_path, dataset)
+
+    entries = []
+    for feature in features:
+        pk = feature[dataset.primary_key]
+        object_path = "/".join([dataset.path, dataset.get_feature_path(pk)])
+        bin_feature = dataset.encode_feature(feature)
+        blob_id = repo.create_blob(bin_feature)
+        entries += [pygit2.IndexEntry(object_path, blob_id, pygit2.GIT_FILEMODE_BLOB)]
```
Comment (craigds) [inline sno/resolve.py:47]:
```suggestion
        entries.append(pygit2.IndexEntry(object_path, blob_id, pygit2.GIT_FILEMODE_BLOB))
```

File: sno/resolve.py (hunk 1 above)
Comment (craigds) [reply sno/resolve.py:29]: although, there's only going to be two features... so... never mind? 😄

Comment (olsen232) [reply sno/resolve.py:29]: Done

File: sno/resolve.py (hunk 2 above)
Comment (olsen232) [reply sno/resolve.py:47]: Done

File: sno/gpkg.py [hunk 3]
```diff
@@ -360,6 +361,18 @@ def ogr_to_gpkg_geom(
     return header + envelope + wkb
 
 
+def json_ogr_to_gpkg_geom(json_ogr, **kwargs):
```
[the remark this answers is not in the document]
Comment (olsen232) [reply sno/gpkg.py:364]: Done

**Pull request merged**